test/e2e/common/pods.go

```diff
@@ -848,4 +853,115 @@ var _ = framework.KubeDescribe("Pods", func() {
+						"image": testPodImage2,
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err, "failed to marshal JSON patch for Pod")
+		_, err = f.ClientSet.CoreV1().Pods(testNs).Patch(context.TODO(), testPodName, types.StrategicMergePatchType, []byte(podPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch Pod")
+
+		ginkgo.By("getting the Pod and ensure it's patched")
+		pod, err := f.ClientSet.CoreV1().Pods(testNs).Get(context.TODO(), testPodName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to fetch Pod")
+		framework.ExpectEqual(pod.ObjectMeta.Labels["test-pod-static"], "true", "failed to patch Pod - missing label")
+		framework.ExpectEqual(pod.Spec.Containers[0].Image, testPodImage2, "failed to patch Pod - wrong image")
+
+		ginkgo.By("getting the PodStatus")
+		podStatusUnstructured, err := dc.Resource(podResource).Namespace(testNs).Get(context.TODO(), testPodName, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err, "failed to fetch PodStatus")
+		podStatusUjson, err := json.Marshal(podStatusUnstructured)
+		framework.ExpectNoError(err, "failed to marshal unstructured response")
+		var podStatus v1.Pod
+		json.Unmarshal(podStatusUjson, &podStatus)
+
+		ginkgo.By("replacing the Pod's status Ready condition to False")
+		podStatusUpdated := podStatus
+		podStatusFieldPatchCount := 0
+		podStatusFieldPatchCountTotal := 2
+		for pos, cond := range podStatusUpdated.Status.Conditions {
+			if (cond.Type == "Ready" && cond.Status == "True") || (cond.Type == "ContainersReady" && cond.Status == "True") {
+				podStatusUpdated.Status.Conditions[pos].Status = "False"
+				podStatusFieldPatchCount++
+			}
+		}
+		framework.ExpectEqual(podStatusFieldPatchCount, podStatusFieldPatchCountTotal, "failed to patch all relevant Pod conditions")
+		podStatusUpdate, err := f.ClientSet.CoreV1().Pods(testNs).UpdateStatus(context.TODO(), &podStatusUpdated, metav1.UpdateOptions{})
+		framework.ExpectNoError(err, "failed to update PodStatus")
+
+		ginkgo.By("check the PodStatus updates's return status to ensure it's Ready condition is False")
+		podStatusFieldPatchCount = 0
+		podStatusFieldPatchCountTotal = 2
+		for _, cond := range podStatusUpdate.Status.Conditions {
+			if (cond.Type == "Ready" && cond.Status == "False") || (cond.Type == "ContainersReady" && cond.Status == "False") {
+				podStatusFieldPatchCount++
+			}
+		}
+		framework.ExpectEqual(podStatusFieldPatchCount, podStatusFieldPatchCountTotal, "failed to update PodStatus - field patch count doesn't match the total")
+
+		ginkgo.By("deleting the Pod via a Collection with a LabelSelector")
+		err = f.ClientSet.CoreV1().Pods(testNs).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "test-pod-static=true"})
+		framework.ExpectNoError(err, "failed to delete Pod by collection")
+
+		ginkgo.By("watching for Pod to be deleted")
+		for watchEvent := range podWatchChan {
+			if watchEvent.Type == "DELETED" {
```

Comment:
the constant should be used:
https://github.com/kubernetes/apimachinery/blob/eca85dcd99946123e54b21f850d895bfdde89d91/pkg/watch/watch.go#L46

Reply:
Updated in 1e98c1ba6c0